server/src/main/java/com/hedera/block/server/verification/VerificationConfig.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.verification;
+
+import com.hedera.block.server.verification.session.BlockVerificationSessionType;
+import com.swirlds.config.api.ConfigData;
+import com.swirlds.config.api.ConfigProperty;
+
+/**
+ * Configuration for the verification module.
+ *
+ * @param enabled whether the verification module is enabled
+ * @param sessionType the type of the verification session
+ * @param hashCombineBatchSize the size of the batch used to combine hashes
+ */
+@ConfigData("verification")
+public record VerificationConfig(
+        @ConfigProperty(defaultValue = "true") boolean enabled,
+        @ConfigProperty(defaultValue = "ASYNC") BlockVerificationSessionType sessionType,
+        @ConfigProperty(defaultValue = "32") int hashCombineBatchSize) {
+
+    /**
+     * Constructs a new instance of {@link VerificationConfig}.
+     */
+    private static final System.Logger LOGGER = System.getLogger(VerificationConfig.class.getName());
+
+    /**
+     * Constructs a new instance of {@link VerificationConfig}.
+     *
+     * @param enabled              whether the verification module is enabled
+     * @param sessionType          the type of the verification session
+     * @param hashCombineBatchSize the size of the batch used to combine hashes
+     */
+    public VerificationConfig {
+        // hashCombineBatchSize must be even and greater than 2
+        if (hashCombineBatchSize % 2 != 0 || hashCombineBatchSize < 2) {
```

Comment:
P.S. also you can have a `requireEven` precondition as well, would be nice. It would call the utility which would return a boolean and based on that you could throw. Just a suggestion.

Reply:
Nice suggestion, and got to learn a lot about the package commons that you added, really curious on how `@ParameterizedTest` works, I think that it has plenty use cases for sure. 👍

Reply:
✅ Done!!